File: python/examples/scatter.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, save_example, scatter_series

META = ExampleMeta(
    slug="scatter",
    title="Scatter plot",
    summary="A scatter plot for irregular point clouds.",
    section="Basic plots",
)


def build_plot():
    x, y = scatter_series()
    return (
        base_plot("Scatter Plot")
        .xlabel("feature")
        .ylabel("response")
        .scatter(x, y)
    )


if __name__ == "__main__":
    save_example(META, build_plot())

File: packages/ruviz/examples/scatter.ts
import { createPlot } from "../src/index.js";
import { scatterSeries } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "scatter",
  title: "Scatter plot",
  summary: "A scatter plot for irregular point clouds.",
  category: "basic",
  createPlot() {
    const { x, y } = scatterSeries();
    return createPlot()
      .sizePx(640, 320)
      .title("Scatter Plot")
      .xlabel("feature")
      .ylabel("response")
      .scatter({ x, y });
  },
};
